StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1210UsingDirectivesMustBeOrderedAlphabeticallyByNamespace.cs

```diff
@@ -50,108 +56,60 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
-            string fullyQualifiedNamespace = namespaceSymbol.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat);
-
-            CompilationUnitSyntax compilationUnit = syntax.Parent as CompilationUnitSyntax;
-            SyntaxList<UsingDirectiveSyntax>? usingDirectives = compilationUnit?.Usings;
-            if (!usingDirectives.HasValue)
-            {
-                NamespaceDeclarationSyntax namespaceDeclaration = syntax.Parent as NamespaceDeclarationSyntax;
-                usingDirectives = namespaceDeclaration?.Usings;
-            }
-
-            if (!usingDirectives.HasValue)
-            {
-                return;
-            }
-
-            foreach (var usingDirective in usingDirectives)
-            {
-                // we are only interested in nodes before the current node
-                if (usingDirective == syntax)
-                {
-                    break;
-                }
+            var compilationUnit = (CompilationUnitSyntax)context.Node;
 
-                // ignore using alias directives, since they are handled by SA1209
-                if (usingDirective.Alias != null)
-                {
-                    continue;
-                }
-
-                INamespaceSymbol precedingNamespaceSymbol;
-                string precedingTopLevelNamespace = GetTopLevelNamespace(semanticModel, usingDirective, out precedingNamespaceSymbol, context.CancellationToken);
-                if (precedingTopLevelNamespace == null || precedingNamespaceSymbol == null)
-                {
-                    continue;
-                }
+            ProcessUsings(compilationUnit.Usings, context);
+        }
 
-                // compare System namespaces to each other, and non-System namespaces to each other
-                if ("System".Equals(precedingTopLevelNamespace, StringComparison.Ordinal) != systemNamespace)
-                {
-                    continue;
-                }
+        private void HandleNamespaceDeclaration(SyntaxNodeAnalysisContext context)
+        {
+            var namespaceDeclaration = (NamespaceDeclarationSyntax)context.Node;
 
-                string precedingFullyQualifiedNamespace = precedingNamespaceSymbol.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat);
-                if (string.Compare(fullyQualifiedNamespace, precedingFullyQualifiedNamespace, StringComparison.OrdinalIgnoreCase) >= 0)
-                {
-                    continue;
-                }
+            ProcessUsings(namespaceDeclaration.Usings, context);
+        }
 
-                string @namespace = namespaceSymbol.ToDisplayString(SymbolDisplayFormat.CSharpErrorMessageFormat);
-                string precedingNamespace = precedingNamespaceSymbol.ToDisplayString(SymbolDisplayFormat.CSharpErrorMessageFormat);
+        private static void ProcessUsings(SyntaxList<UsingDirectiveSyntax> usings, SyntaxNodeAnalysisContext context)
```

Comment:
I would swap method arguments. Implemented rules in project  that I've seen before accepts `context` as first argument. I think we should be consistent in such things.